core/domain/collection_validators_test.py

```diff
@@ -0,0 +1,1638 @@
+            (
+                u'[u\'failed validation check for collection_ids '
+                'field check of CollectionCommitLogEntryModel\', '
+                '[u"Entity id collection-0-1: based on field collection_ids '
+                'having value 0, expected model CollectionModel with id 0 '
+                'but it doesn\'t exist", u"Entity id collection-0-2: based '
+                'on field collection_ids having value 0, expected model '
+                'CollectionModel with id 0 but it doesn\'t exist"]]'
+            ), u'[u\'fully-validated CollectionCommitLogEntryModel\', 3]']
+        self.run_job_and_check_output(
+            expected_output, literal_eval=True)
+
+    def test_missing_collection_rights_model_failure(self):
+        collection_models.CollectionRightsModel.get_by_id('1').delete(
+            feconf.SYSTEM_COMMITTER_ID, '', [])
+        expected_output = [
+            (
+                u'[u\'failed validation check for collection_rights_ids '
+                'field check of CollectionCommitLogEntryModel\', '
+                '[u"Entity id rights-1-1: based on field '
+                'collection_rights_ids having value 1, expected model '
+                'CollectionRightsModel with id 1 but it doesn\'t exist"]]'
+            ), u'[u\'fully-validated CollectionCommitLogEntryModel\', 3]']
+        self.run_job_and_check_output(
+            expected_output, sort=True)
+
+    def test_invalid_collection_version_in_model_id(self):
+        model_with_invalid_version_in_id = (
+            collection_models.CollectionCommitLogEntryModel.create(
+                '0', 3, self.owner_id, 'edit', 'msg', [{}],
+                constants.ACTIVITY_STATUS_PUBLIC, False))
+        model_with_invalid_version_in_id.collection_id = '0'
+        model_with_invalid_version_in_id.put()
+        expected_output = [
+            (
+                u'[u\'failed validation check for collection model '
+                'version check of CollectionCommitLogEntryModel\', '
+                '[u\'Entity id %s: Collection model corresponding '
+                'to id 0 has a version 1 which is less than '
+                'the version 3 in commit log entry model id\']]'
+            ) % (model_with_invalid_version_in_id.id),
+            u'[u\'fully-validated CollectionCommitLogEntryModel\', 4]']
+        self.run_job_and_check_output(
+            expected_output, sort=True, literal_eval=False)
+
+    def test_model_with_invalid_id(self):
+        model_with_invalid_id = (
+            collection_models.CollectionCommitLogEntryModel(
+                id='invalid-0-1',
+                user_id=self.owner_id,
+                commit_type='edit',
+                commit_message='msg',
+                commit_cmds=[{}],
+                post_commit_status=constants.ACTIVITY_STATUS_PUBLIC,
+                post_commit_is_private=False))
+        model_with_invalid_id.collection_id = '0'
+        model_with_invalid_id.put()
```

Comment:
@KhushiGangopadhyay, I think you have copied the old test code here.PTAL! ([link](https://github.com/oppia/oppia/pull/11152/files#diff-d1da302dac071f9539475f0878c01e0639aafad1e965b9437ccff81ec6210583L1590)) 

[If it's getting complicated then try reworking on this on an new updated branch!]

Reply:
Yes, I understand now, I'll fix it